### ISSUE TITLE
Eliminate Client and Server Builders

### ISSUE THREAD

**Issue by NiteshKant** [ReactiveX/RxNetty#280]
Today, all clients and servers follow the builder pattern to make sure that a client/server once created in immutable. In order to make the creation of most common clients/server easy, `RxNetty` factory class has some factory methods, like:

``` java
    public static RxServer<ByteBuf, ByteBuf> createTcpServer(final int port,
                                                             ConnectionHandler<ByteBuf, ByteBuf> connectionHandler);

    public static HttpServer<ByteBuf, ByteBuf> createHttpServer(int port, RequestHandler<ByteBuf, ByteBuf> requestHandler);
```

It also has the following builder creation methods:

``` java
public static <I, O> ServerBuilder<I, O> newTcpServerBuilder(int port, ConnectionHandler<I, O> connectionHandler);
public static <I, O> HttpClientBuilder<I, O> newHttpClientBuilder(String host, int port);
```

The above two concepts are a bit odd as a user has to unlearn one to learn the other. eg: If one needs to enable wire-level logging, there is no way, one can do that while creating a client/server. One, has to use the builder.
### Proposal
- Fold all builder methods into the main client/server. Every invocation of those methods, should create a new instance of client/server.
- Replace inheritance with delegation in clients/servers. In absence of this change, the client/server have to introduce a generic parameter which will guide what to return from these methods which create a new client/server. The downside is that an `HttpClient` can not be used as an `RxClient`. The usage of this aspect is very limited as once, a user uses an `HttpClient`, there is very little use of using `#connect()` on the client.

After this change, users of a particular client/server would just have to be aware of any APIs on that client/server and nothing else. There will be less abstractions and less confusion.

**Comment by benjchristensen** [comment]
Can you provide examples of what the new API would be and code examples of using them?

**Comment by NiteshKant** [comment]
#303 details the API changes discussed here with examples.

**Comment by dimitarg** [comment]
I guess the downside of this approach is that once one is done with
configuring the server, all these methods you never need to use anymore get
in the way while you're interacting with the built server. In academic
speak it violates SRP. But I'm not sure how to improve this while
addressing your original concerns.
As an end user I like it one way or the other, usually my building code
tends to be about 10 lines in one place in the app. Still I wanted to point
out.

**Comment by NiteshKant** [comment]
@dimitarg Thanks for sharing your views, much appreciated!

Yeah the classes are bulky but at the end of the day the intention is to remove multiple abstraction and streamline the usage so that it is intuitive to use a client/server which it currently isn't IMO.

**Comment by NiteshKant** [comment]
Available in 0.5.x branch
